actions/arteria_delivery_service.py

```diff
@@ -86,21 +87,48 @@ class ArteriaDeliveryQuerier(ArteriaQuerierBase):
     delivery_in_progress = 'delivery_in_progress'
     delivery_successful = 'delivery_successful'
     delivery_failed = 'delivery_failed'
+    delivery_skipped = 'delivery_skipped'
+
+    valid_states = [pending, mover_processing_delivery, mover_failed_delivery,
+                    delivery_in_progress, delivery_successful, delivery_failed]
+
+    successful_states = delivery_successful
 
     def __init__(self, logger):
         super(ArteriaDeliveryQuerier, self).__init__(logger)
 
     def successful_status(self):
-        return self.delivery_successful
+        return self.successful_states
 
     def failed_status(self):
         return [self.mover_failed_delivery, self.delivery_failed]
 
     def valid_status(self):
-        valid_states = [self.pending, self.mover_processing_delivery, self.mover_failed_delivery,
-                        self.delivery_in_progress, self.delivery_successful, self.delivery_failed]
-        return valid_states
-
+        return self.valid_states
+
+    def query_for_status(self, link, skip_mover):
+        if skip_mover:
+            self.valid_states += self.delivery_skipped
+            self.successful_states = self.delivery_skipped
+
+        while True:
+            response = requests.get(link)
+            response_as_json = json.loads(response.content)
+            status = response_as_json["status"]
+
+            if status == self.successful_status():
+                self.logger.info("Got successful status {}".format(status))
+                return True
+            elif status in self.failed_status():
+                self.logger.warning("Got unsuccessful status {}".format(status))
+                return False
+            elif status in self.valid_status():
+                self.logger.info("Got valid, but not final status, status: {}. Will keep polling.".format(status))
+                # TODO Make timeout configurable
+                time.sleep(5)
```

Comment:
I presume you want to have such an aggressive timeout for testing purposes/fast iteration. When we have it configurable I'd say we'd like to wait a bit more. :)